beacon_node/lighthouse_network/src/service/mod.rs

```diff
@@ -353,34 +353,71 @@ impl<AppReqId: ReqId, TSpec: EthSpec> Network<AppReqId, TSpec> {
             }
         };
 
-        let (swarm, bandwidth) = {
-            // Set up the transport - tcp/ws with noise and mplex
-            let (transport, bandwidth) = build_transport(local_keypair.clone())
-                .map_err(|e| format!("Failed to build transport: {:?}", e))?;
-
-            // use the executor for libp2p
-            struct Executor(task_executor::TaskExecutor);
-            impl libp2p::swarm::Executor for Executor {
-                fn exec(&self, f: Pin<Box<dyn futures::Future<Output = ()> + Send>>) {
-                    self.0.spawn(f, "libp2p");
-                }
+        struct Executor(task_executor::TaskExecutor);
+        impl libp2p::swarm::Executor for Executor {
+            fn exec(&self, f: Pin<Box<dyn futures::Future<Output = ()> + Send>>) {
+                self.0.spawn(f, "libp2p");
             }
+        }
 
-            // sets up the libp2p connection limits
-
-            (
-                SwarmBuilder::with_executor(
-                    transport,
-                    behaviour,
-                    local_peer_id,
-                    Executor(executor),
-                )
-                .notify_handler_buffer_size(std::num::NonZeroUsize::new(7).expect("Not zero"))
-                .per_connection_event_buffer_size(4)
-                .build(),
-                bandwidth,
-            )
-        };
+        let (swarm_builder, bandwidth) =
+            libp2p::SwarmBuilder::with_existing_identity(local_keypair.clone())
+                .with_tokio()
+                // Using `with_other_transport` in order to support mplex. In case mplex would not be needed, the below would be as simple as:
```

Comment:
How hard would it be to support a custom encryption protocol in the builder?

Reply:
You mean a custom multiplexer? Possible. But I don't expect us to write a new multiplexer for TCP any time soon. Thus I would rather like to add such option once it is needed.

In this particular case, Ethereum is planning to move away from mplex. See https://github.com/ethereum/consensus-specs/issues/3490

Reply:
> You mean a custom multiplexer?

Yes.

> But I don't expect us to write a new multiplexer for TCP any time soon. Thus I would rather like to add such option once it is needed.

But it would support these usecases and allow us to eventually deprecate the other upgrade path. I really do not want us to introduce more ways of doing things without deprecating old ones.